src/processMedia.js

```diff
@@ -0,0 +1,23 @@
+import * as cheerio from 'cheerio';
+import fetch from 'node-fetch';
+import processGist from './processors/processGist';
+
+const processMedia = url => new Promise((resolve, reject) => {
+  fetch(url)
+    .then(resp => resp.text())
+    .then((html) => {
+      const $ = cheerio.load(html);
+      if ($('html').has('script[src^="https://gist.github.com/"]').length) { // Github Gist
+        // Extract the Gist's ID:
+        const gistId = $('body > script').attr('src').replace(/^.*[\\\/]/, '').replace(/\.js$/, '');
+        processGist(gistId)
+          .then((markdown) => { resolve(markdown); });
+      } else {
+        // Resolve the empty string with a line break if the medium type is not known
+        resolve('\n');
+      }
+    })
+    .catch(err => reject(err));
```

Comment:
ugh :stuck_out_tongue_closed_eyes:

Reply:
@gunar sorry this one must have slipped through the cracks in the early hours one morning 😝 What is best to do here in your opinion? Just ignore the caught error and just add a blank new line where the async media could not be fetched?

e.g:

```js
.catch(err => {
  // log err somewhere, then:
  resolve('\n');
})
```